### PR TITLE
Update API to OSBA & library dependencies

### PR DESCRIPTION
**Pull request by spiegela**
Upgraded compile dependencies:
* Spring Boot 2.2.5
* Spring Gradle Dependency Management Plugin:  1.0.9
* Spring Cloud Service Broker: 3.1.0
* ECS Object Client: 3.1.3

New compile dependency:
* Google Guava: 28.2

Updated development dependencies:
* Wiremock: 2.26.1
* Powermock API Mockito2: 2.0.5
* Powermock JUnit4: 1.7.1
* Ginkgo4j: 1.0.11
* Gradle[w]: 6.2.1

This change alters the logic to get retrieve the namespace URL simplifying the workflow by retrieving service settings directly from the associated service-instance, instead of looking up service/plan from the catalog.

This change includes Helm chart updates to support new OSBA in K8s installations provided by @maddisondavid.

### CODE REVIEW

File: src/main/java/com/emc/ecs/servicebroker/service/BindingWorkflowImpl.java [hunk 1]
```diff
@@ -25,6 +25,7 @@
     BindingWorkflowImpl(ServiceInstanceRepository instanceRepo, EcsService ecs) {
         this.instanceRepository = instanceRepo;
         this.ecs = ecs;
+        this.service = service;
```
Comment (maddisondavid) [inline src/main/java/com/emc/ecs/servicebroker/service/BindingWorkflowImpl.java:28]: Value is assigned to itself and not passed in as a parameter

Comment (rogalv) [reply src/main/java/com/emc/ecs/servicebroker/service/BindingWorkflowImpl.java]: Removed that line.
Btw, ServiceDefinitionProxy service seems to be unused now. Remove it too?

File: charts/templates/configmap.yaml [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
-      namespace: "131118670375936839"
-      replication-group: ecstestdrivegeo
-      prefix: kubetesting-
+      object-endpoint: {{ .Values.api.endpoint }}
+      management-endpoint: {{ .Values.ecsConnection.endpoint }}
+      namespace: {{ .Values.namespace }}
+      replication-group: {{ .Values.replicationGroup }}
+      prefix: {{ .Values.prefix }}
+  {{- if .Values.certificate }}
+    certificate: {{ toYaml .Values.certificate | indent 6 }}
```
[the remark this answers is not in the document]
Comment (rogalv) [reply charts/templates/configmap.yaml:15]: Indent value should be increased too then.

[the remark this answers is not in the document]
Comment (maddisondavid) [reply charts/templates/configmap.yaml:15]: Hmm, the indent SEEMS fine when I've tested it, it's just that the `certificate` element was not at the right level.

Just indenting `certifacate` in by two spaces gives me the result that I need in the resulting `ConfigMap`

```
  application.yml: |-
    ---
    broker:
      object-endpoint: https://object.ecstestdrive.com
      management-endpoint: https://portal.ecstestdrive.com
      namespace: 131118670375936839
      replication-group: ecstestdrivegeo
      prefix: kubetesting-
      certificate:       |-
        -----BEGIN CERTIFICATE-----
        MIIDCTCCAfGgAwIBAgIJAJ1g36y+tM0RMA0GCSqGSIb3DQEBCwUAMBQxEjAQBgNV
        BAMTCWxvY2FsaG9zdDAeFw0yMDAyMTkxOTMzMjVaFw0zMDAyMTYxOTMzMjVaMBQx
        EjAQBgNVBAMTCWxvY2FsaG9zdDCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoC
        ggEBAKdvGtq912EKCBNsHTYjtbRlDkJyietThHw7p3WQCUq+TBxSisPJCshWRbNr
        vpHjygnwZOSUcmvGdjpgXZeSUTWP0jgjCaOvhgeTHCRTKpkOd+tjEFpxCDGjZ61a
        wHWSzxHKKBCgd8v6ERgkUyaD7S6RaNVjZ6E49raiDqeGmTd1uQRVq8RzRy2HePiT
        G+rSRfS8qP3QzAjw1pIeJU100Ihy6wmMhUhmNJlU93INziF5mg0VYbnSW/bRb7zL
        tNVHI0jkgPDQvqPtf26TZAhIO1bBiWebEk9UUhzGtaNgFE6/k9NTpu7Z32zkvOFe
        vu7lw0x2c24p5mkPedycrGOGkHsCAwEBBaNeMFwwHwYDVR0jBBgwFoAU2us7uV9T
        ZvUaai/9XGPrjhRbKNswGgYDVR0RBBMwEYIJbH9jYWxob3N0hwR/AAABMB0GA1Ud
        DgQWBBTa6zu5X1Nm9RpqL/1cY+uOFFso2zANBgkqhkiG9w0BAQsFAAOCAQEAWcrd
        +6xY62Ctn+ijphaUOuH9GO1VCUjLRPwZc1/QfMLsss/UhJVITt7JhMMUjGcV5UyV
        JbQwoA+vso2DUbfehTceCWOa7QXLnO7si+ujVn0/vlCLvOj7MSJeRwOD3+3b2MpV
        AJImqfo4J7ovY4iYfKPLVO6l1B5N9Cw3F3ztcA3HfmtKS7pTA0H/TLxaJ8GVCQO+
        d+StKXhM4dLC/uyvkPRQhXifMYJB5WM5G1X+kaiD30kZ5R0oEyWmaDrkuy2Xy0AP
        4xCeTPYMHUQlqaUYYG4iNnMfdwJuBiTpEpxKN/zt1M+ycWXSyGSW/EkItxztO49v
        kQjNgkCo/artraBjvQ==
        -----END CERTIFICATE-----
```

File: src/main/java/com/emc/ecs/servicebroker/service/BindingWorkflowImpl.java (hunk 1 above)
Comment (maddisondavid) [reply src/main/java/com/emc/ecs/servicebroker/service/BindingWorkflowImpl.java]: I think it's used in `getCredentials`
```
        String endpoint = ecs.getNamespaceURL(ecs.prefix(namespaceName), service, plan,
                createRequest.getParameters());
```

Comment (rogalv) [reply src/main/java/com/emc/ecs/servicebroker/service/BindingWorkflowImpl.java]: I can't find that line in current branch.
Anyway, it's ok for the moment.

**Pull request merged**